academy/exchange/cloud/config.py

```diff
@@ -0,0 +1,154 @@
+"""Cloud exchange configuration file parsing."""
+
+from __future__ import annotations
+
+import logging
+import pathlib
+import sys
+from typing import Any
+from typing import BinaryIO
+from typing import Dict  # noqa: UP035
+from typing import Literal
+from typing import Optional
+from typing import TypeVar
+from typing import Union
+
+if sys.version_info >= (3, 11):  # pragma: >=3.11 cover
+    from typing import Self
+else:  # pragma: <3.11 cover
+    from typing_extensions import Self
+
+from pydantic import BaseModel
+from pydantic import ConfigDict
+from pydantic import Field
+
+if sys.version_info >= (3, 11):  # pragma: >=3.11 cover
+    import tomllib
+else:  # pragma: <3.11 cover
+    import tomli as tomllib  # type: ignore[import-not-found,unused-ignore]

```

Comment:
Probably need to add the following to the `pytproject.toml` requirements (something else might already require it, but I'd rather be explicit).
```
"tomli ; python_version<'3.11'",
```
and also to the mypy additional requirements in `.pre-commit.config.yaml` so it can get type checked.